### PR TITLE
Extending CI 

### PR DESCRIPTION
**Pull request by scarrazza**
Extends CI to other operating systems. It seems that the cupy does not offer precompiled versions for macos. Unfortunately the pure `cupy` package is not simple to install on CI.

I believe we should consider removing cupy from the requirements of macos builds. I believe the code will crash if cupy is not installed, @stavros11 could you please have a look?

### CODE REVIEW

**Comment by stavros11** [comment]
> Extends CI to other operating systems. It seems that the cupy does not offer precompiled versions for macos. Unfortunately the pure `cupy` package is not simple to install on CI.

I don't think we need to install cupy on CI for any operating system. CI machines do not have GPUs anyway so the cupy backend will not be tested even if we install it.

> I believe we should consider removing cupy from the requirements of macos builds. 

Personally I would be okay even with removing cupy from qibojit's requirements completely. This is mostly because it is safer if cupy is installed seperately in order to match each machine's cuda version. At least this is what I do when I install qibojit. In this case we should include in the docs that cupy has to be installed seperately.

> I believe the code will crash if cupy is not installed, @stavros11 could you please have a look?

All cupy imports are inside the `CupyBackend` class which is initialized only if cupy is available so the code should not crash. It should be possible to use qibojit in a machine without GPU and cupy, using the numba part only. I tested in a clean environment without cupy in my machine using only numba and qibojit installed with `--no-deps` and tests pass. Do you get a crash somewhere if cupy is missing?

**Comment by scarrazza** [comment]
@stavros11 ok, sounds reasonable, we can drop cupy and extend the docs. CI is complaining about pylint.

**Comment by scarrazza** [comment]
This is ready for a quick look before making the first release.

**Review by stavros11 — APPROVED** [review]
Thanks, looks good to me.

File: setup.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from setuptools import setup, find_packages
 import os
 import re
+import sys
```
[the remark this answers is not in the document]
Comment (scarrazza) [reply setup.py]: Thanks!

**Pull request merged**